frontend/src/components/Table/TableComponents/index.tsx

```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { Box } from "@mui/material";
+import { GridColDef, GridToolbarQuickFilter, GridRowsProp } from "@mui/x-data-grid";
+import { TableTotalFooter } from "./TotalFooter";
+
+/**
+ * Formats data from a database table in a way that is usable for material ui datagrid (table).
+ *
+ * @param   {string[] | undefined}                               totalColumns totalColumns to show totals for money columns.
+ * @param   {GridRowsProp}                                       rows         rows used in datagrid.
+ * @param   {GridColDef[]}                                       columns      columns used in datagrid.
+ * @returns { Toolbar: quickSearchToolbar, Footer: tableFooter }              an object with that contains custom components for the table
+ */
+
+export const TableComponents = (
+  totalColumns: string[] | undefined,
+  rows: GridRowsProp,
+  columns: GridColDef[]
+) => {
+  const totals: Array<{ id: string; total: number }> = [];
+  if (totalColumns && rows.length > 0) {
+    totalColumns.forEach((col: string) => {
+      totals.push({
+        id: col,
+        total: rows
+          .map((x: { [x: string]: number }) => x[col])
```

Comment:
also is the output type number for the .map() if it doesn't coerce the string during the map operation?

Reply:
I see you're coercing each row string value to a number during the reduce

So I figure it might be simpler to do that during the `.map()`

and also the map seems to contain a typescript notation that talks about converting 'x' (a string) to 'x' (a number), but the .map() doesn't do that.